File: core/src/net/natruid/jungle/components/render/PivotComponent.java
package net.natruid.jungle.components.render;

public class PivotComponent extends Vector2Component {
    public static final PivotComponent DEFAULT = new PivotComponent();

    public PivotComponent() {
        this(.5f, .5f);
    }

    public PivotComponent(float x, float y) {
        super(x, y);
    }

    @Override
    protected void reset() {
        set(DEFAULT);
    }
}

File: core/src/net/natruid/jungle/components/render/PivotComponent.kt
package net.natruid.jungle.components.render

class PivotComponent(x: Float = 0.5f, y: Float = 0.5f) : Vector2Component(x, y) {
    companion object {
        val DEFAULT = PivotComponent()
    }

    override fun reset() {
        set(0.5f, 0.5f)
    }
}
